rfd/0131-adminitrative-actions-mfa.md

```diff
@@ -0,0 +1,274 @@
+---
+authors: Brian Joerger (bjoerger@goteleport.com)
+state: draft
+---
+
+# RFD 131 - Administrative Actions MFA
+
+## Required Approvers
+
+- Engineering: @r0mant && @codingllama
+- Product: @klizhentas || @xinding33
+- Security: @jentfoo
+
+## What
+
+Enforce MFA verification for any administrative actions taken. For example:
+
+- Adding, removing, or editing Nodes and other services.
+- Adding, removing, or editing users, roles, auth connectors, and other
+administrative resources.
+- Approving access requests.
+
+This will apply to administrative actions performed via the WebUI, `tctl`,
+and `tsh`, or Teleport Connect.
+
+## Why
+
+Currently, Teleport only prompts for MFA verification in a few circumstances:
+
+1. User logs in with MFA.
+2. User updates their password or registered MFA devices.
+3. User connects to a service that requires Per-session MFA.
+
+As a result, the vast majority of Teleport actions do not require timely MFA
+verification. If an admin's login session is compromised, it can be used to
+perform attacks pertaining to privilege escalation, security downgrades, Auth
+connector changes, and other forms of cluster sabotage.
+
+Adding an MFA restriction to administrative actions limits this attack vector
+by re-verifying the user's identity promptly before performing any administrative
+action that could be abused.
+
+## Details
+
+### Server configuration
+
+Starting in Teleport 14, MFA for administrative actions will be turned on by
+default for all users with an MFA device registered. This means that when
+a cluster's second factor is set to one of `optional`, `on`, `webauthn`, or
+`otp`, all normal login sessions will require MFA verification to complete
+admin actions.
+
+### Administrative Actions
```

Comment:
Also, even with these examples tbh I'm having a bit of a hard time tracking what _exactly_ will require MFA and what will not. I think the RFD should specify explicitly and exhaustively the actions that will be protected and which won't. I understand it might be quite a bit of work to collect those but I'm not sure how otherwise we can be sure we haven't missed anything.